src/main/java/com/bento909/FileBasedLeagueGenerator.java

```diff
@@ -56,4 +50,22 @@ private Match generateMatch(List<String> fileLine) throws ParseException {
                 .withManOfTheMatch(fileLine.get(MAN_OF_THE_MATCH_INDEX))
                 .build();
     }
+
+    private Map<String, List<Match>> addTeamToTeamMap (Map<String, List<Match>> teamMap, String team, Match match) {
+        if (!teamMap.containsKey(team)) {
+            teamMap.put(team, new ArrayList<>());
+        }
+        teamMap.get(team).add(match);
+        return teamMap;
+    }
+
+    private TreeSet<Team> addTeamsToTreeSet (Map<String, List<Match>> teamMap) {
+        TreeSet<Team> teams = new TreeSet<>();
```

Comment:
See if you can make this an immutable set.

https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/ImmutableSet.html

You will need to use the Builder for the class